components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -886,8 +964,17 @@ private void updateMetadataForMYSQL(Resource resource, String resourceTypeId, bo
                             not create files. It is allowed to create a resource without files or attributes in order
                             to allow  file upload after resource creation.
                             */
-                            preparedStatement.setBoolean(++initialParameterIndex, false);
-                            preparedStatement.setBoolean(++initialParameterIndex, isAttributeExists);
+                            try {
+                                if (isOracleDB() || isMSSqlDB()) {
+                                    preparedStatement.setInt(++initialParameterIndex, 0);
+                                    preparedStatement.setInt(++initialParameterIndex, isAttributeExists ? 1 : 0);
+                                } else {
+                                    preparedStatement.setBoolean(++initialParameterIndex, false);
+                                    preparedStatement.setBoolean(++initialParameterIndex, isAttributeExists);
+                                }
+                            } catch (DataAccessException e) {
+                                throw new RuntimeException(e);
```

Comment:
add a descripive error message to the exception

Reply:
addressed in b9d69a1ace1fb2380564347a0b2eaa96eb14a912